ArduCopter/mode_auto.cpp

```diff
@@ -324,32 +324,32 @@ void ModeAuto::takeoff_start(const Location& dest_loc)
     // by default current_alt_cm and alt_target_cm are alt-above-EKF-origin
     int32_t alt_target_cm;
     bool alt_target_terrain = false;
-    float current_alt_cm = inertial_nav.get_position_z_up_cm();
+    int32_t current_alt_cm = static_cast<int32_t>(inertial_nav.get_position_z_up_cm());
```

Comment:
any reason why we're changing this to an int32?

Reply:
final use of this is as an int. The only place we need the float version is L335 with the terrain offset. Using directly an int and casting the terrain_offset (in cm , so we don't lose accurracy) save us 16bytes

Reply:
Keep as float please.

Reply:
it is useless to keep it as float since all use from this variable are implicitly casting to int32_t, unless you got some WIP on those files